### PR TITLE
e2e: script check testing

### PR DESCRIPTION
**Pull request by tgross**
In support of #6197 and #6162

### CODE REVIEW

File: e2e/consul/script_checks.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+	jobIds []string
+}
+
+func init() {
+	framework.AddSuites(&framework.TestSuite{
+		Component:   "ScriptChecks",
+		CanRunLocal: true,
+		Consul:      true,
+		Cases: []framework.TestCase{
+			new(ScriptChecksE2ETest),
```
Comment (schmichael) [inline e2e/consul/script_checks.go]: You could also add this to the `init` func in `consul.go` if you want to skip some boilerplate.

File: e2e/consul/script_checks.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+	jobId := "checks_group" + uuid[0:8]
+	tc.jobIds = append(tc.jobIds, jobId)
+
+	// Job run: verify that checks were registered in Consul
+	allocs := e2eutil.RegisterAndWaitForAllocs(f.T(),
+		nomadClient, "consul/input/checks_group.nomad", jobId)
+	require.Equal(1, len(allocs))
+	require.Eventually(func() bool {
+		serviceStatus(require, consulClient, "group-check-1", capi.HealthPassing)
+		// TODO(tgross): we want to verify this goes from unhealthy to
```
Comment (tgross) [inline e2e/consul/script_checks.go]: I think I've figured out what I'll do here: for the `Crash a task: verify that checks become unhealthy and then healthy` test, I need to reach into the task (probably via nomad exec). So I'll update one of these checks to read a non-existent file in the task, verify Consul sees it as unhealthy, nomad exec into the task to touch the file, and that'll cause the health check to flip to healthy.

**Comment by tgross** [comment]
These are passing vs a local environment that includes #6265:

```
▶ go test -v ./e2e/ -run 'TestE2E/Consul/\*consul\.ScriptChecksE2ETest'
=== RUN   TestE2E
=== RUN   TestE2E/Consul
=== RUN   TestE2E/Consul/*consul.ScriptChecksE2ETest
=== RUN   TestE2E/Consul/*consul.ScriptChecksE2ETest/TestGroupScriptCheck
=== RUN   TestE2E/Consul/*consul.ScriptChecksE2ETest/TestTaskScriptCheck
=== RUN   TestE2E/Consul_Template
--- PASS: TestE2E (65.71s)
    --- PASS: TestE2E/Consul (65.62s)
        --- PASS: TestE2E/Consul/*consul.ScriptChecksE2ETest (65.45s)
            --- PASS: TestE2E/Consul/*consul.ScriptChecksE2ETest/TestGroupScriptCheck (33.17s)
            --- PASS: TestE2E/Consul/*consul.ScriptChecksE2ETest/TestTaskScriptCheck (32.15s)
    --- PASS: TestE2E/Consul_Template (0.09s)
PASS
ok      github.com/hashicorp/nomad/e2e  68.029s
```

**Comment by tgross** [comment]
With the fix for macOS vs Linux exit codes in f856cd4 and running against a build that includes #6291, this is now passing!

```
▶ go test -v ./e2e/ -run 'TestE2E/Consul/\*consul\.ScriptChecksE2ETest'
=== RUN   TestE2E
=== RUN   TestE2E/Consul
=== RUN   TestE2E/Consul/*consul.ScriptChecksE2ETest
=== RUN   TestE2E/Consul/*consul.ScriptChecksE2ETest/TestGroupScriptCheck
=== RUN   TestE2E/Consul/*consul.ScriptChecksE2ETest/TestTaskScriptCheck
=== RUN   TestE2E/Consul_Template
--- PASS: TestE2E (75.87s)
    --- PASS: TestE2E/Consul (75.79s)
        --- PASS: TestE2E/Consul/*consul.ScriptChecksE2ETest (75.63s)
            --- PASS: TestE2E/Consul/*consul.ScriptChecksE2ETest/TestGroupScriptCheck (38.20s)
            --- PASS: TestE2E/Consul/*consul.ScriptChecksE2ETest/TestTaskScriptCheck (37.30s)
    --- PASS: TestE2E/Consul_Template (0.09s)
PASS
ok      github.com/hashicorp/nomad/e2e  78.862s
```

**Pull request merged**